### PR TITLE
bldy: envinroment variables can now be used in paths

### PR DESCRIPTION
**Pull request by sevki**
related to bldy/build#45

Signed-off-by: Sevki <s@sevki.org>

### CODE REVIEW

File: sys/src/libthread/BUILD [hunk 1]
```diff
@@ -8,7 +8,7 @@ cc_library(
             "//amd64/include",
         ],
 	srcs = [
-		"amd64.c",
+		"$ARCH.c",
```
Comment (rminnich) [inline sys/src/libthread/BUILD]: Do you need the ${ARCH} form here? Just wondering.

File: .build [hunk 2]
```diff
@@ -1,4 +1,5 @@
 CC = gcc
```
Comment (rminnich) [inline .build]: I'm not a huge fan of persistent .files. Is this necessary or could we just make it a real name?

Comment (sevki) [inline .build]: it's not at all necessary but I really don't want to be typing `ARCH=amd64 bldy //:kernel` everytime. contents of this file is overwritten if you specify a real env var like so `ARCH=risc-v bldy //:kernel` bldy will overwrite the ARCH set in .file

File: sys/src/libthread/BUILD (hunk 1 above)
Comment (sevki) [inline sys/src/libthread/BUILD]: I was hoping you'd ask about this, if it's a `.` or `/` terminated, it doesn't seem like you have to. but I can change this to that form for persistence if you'd like.

File: .build (hunk 2 above)
Comment (rminnich) [inline .build]: why a .file however? That stuff is a cancer. Why not a file named ENV or DEFAULTS or something reasonably official?

overall however I am not convinced this is a good idea. It's not that hard to
export ARCH=amd64
and then not worry. But it's your tool, up to you, that's why I approved it.
Also, your .build hardwires amd64 and in general hardwire ARCH in any file is bad policy, especially one checked in to git.

Comment (sevki) [inline .build]: it's not my tool, that's why I created a new org (which is eagerly awaiting you to join) and invited you all.
the .build file comes from the buck and bazel family, I really didn't think too much over it. I'll create an issue and change the .build to something like `bldyconfig`

Comment (rminnich) [inline .build]: Thank! I really appreciate your taking a look at this!

You can go ahead and commit this PR if you want, the other change can come later, OR you can make the change now. I leave it in your hands :-)

Comment (sevki) [inline .build]: Ok, I'll merge this now. I have more related to this that will come in but I'd rather avoid making this a cathedral

File: sys/src/libthread/BUILD (hunk 1 above)
Comment (rminnich) [inline sys/src/libthread/BUILD]: I'd suggest making it always work one way. We can see that went really well with Go, people always know there's one and only one way to do things.

Comment (sevki) [inline sys/src/libthread/BUILD]: should we check to see if it matches `${.*}` and complain if it doesn't?

Comment (rminnich) [inline sys/src/libthread/BUILD]: That regexp is not safe but if you're saying that a $ should be followed by a single {, a set of valid non-} terminated by a }, then you're ok I guess.
${.*} matches ${{{{{{{{{{{{{{{{{{{{{{{{{{{} for example. :-)

Comment (sevki) [inline sys/src/libthread/BUILD]: well my question was more about whether we should keep the go functionality as defined [here](https://golang.org/src/os/env.go?s=378:435#L3) or be more opinionated. I'll change it to`${[A-Z]*}`

Comment (rminnich) [inline sys/src/libthread/BUILD]: Sorry! I misunderstood.

Comment (sevki) [inline sys/src/libthread/BUILD]: no bother, what's your take on being more opinionated?

Comment (rminnich) [inline sys/src/libthread/BUILD]: ??

People need to be opinionated :-)

**Pull request merged**